### PR TITLE
Added new 'jerry_value_instanceof' API function.

### PR DESCRIPTION
**Pull request by LaszloLango**
JerryScript-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -1084,6 +1084,84 @@ jerry_get_global_object (void);
...
+
+```c
+jerry_value_t
+jerry_value_instanceof (const jerry_value_t value, const jerry_value_t prototype);
+```
+
+- `value` - api value of instance object
+- `prototype` - api value of prototype
+- return value
+  - error, if value argument is not an object or prototype argument is not a function
```
Comment (galpeter) [inline docs/02.API-REFERENCE.md]: There should be a test where the input `value` (and also for the `prototype`) is not an object to trigger this case.

File: docs/02.API-REFERENCE.md [hunk 2]
```diff
@@ -1084,6 +1084,84 @@ jerry_get_global_object (void);
 
 Functions to check the type of an API value ([jerry_value_t](#jerry_value_t)).
 
+
+## jerry_value_instanceof
+
+**Summary**
+
+Returns whether the given `jerry_value_t` is an instance of the given prototype.
```
Comment (galpeter) [inline docs/02.API-REFERENCE.md]: Isn't there any difference between the es5.1 vs es2015 `instanceof` internal workings? Could that affect the api workings?

Comment (LaszloLango) [reply docs/02.API-REFERENCE.md]: It should not affect the api workings, but may affect the internal workings (e.g.: instance of classes).

Comment (rerobika) [reply docs/02.API-REFERENCE.md]: Classes are simple functions, the only specializations they have is a bytecode status flag, so I think it couldn't have any affect on it.

File: jerry-core/api/jerry.c [hunk 3]
```diff
@@ -601,6 +601,34 @@ jerry_get_global_object (void)
...
+                        const jerry_value_t prototype) /**< prototpye function */
+{
+  jerry_assert_api_available ();
+
+  if (ecma_is_value_error_reference (value))
+  {
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (error_value_msg_p)));
+  }
+
+  ecma_object_t *proto_obj_p = ecma_get_object_from_value (prototype);
```
Comment (rerobika) [inline jerry-core/api/jerry.c]: Couldn't it cause an assertion fail if this function is called like `jerry_value_instanceof (jerry_create_number (5), jerry_create_number (5))`, due to the second argument is not an object?

File: tests/unit-core/test-api-instaceof.c [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+{
+  (void) func_val;
+  (void) this_val;
+  (void) argv;
+  (void) argc;
+  return jerry_create_undefined ();
+} /* my_constructor */
+
+int
+main (void)
```
Comment (galpeter) [inline tests/unit-core/test-api-instaceof.c]: There should be more test cases, for example:
* string, string
* string, number
* array, array
* array, object
* object, array
* null, object
* object, string
* typedarray, object
* arraybuffer, object
* symbols, object

File: jerry-core/api/jerry.c [hunk 5]
```diff
@@ -601,6 +601,40 @@ jerry_get_global_object (void)
...
+ * Note:
+ *      returned value must be freed with jerry_release_value, when it is no longer needed.
+ *
+ * @return true  - if the specified value is an instance of the given prototype
+ *         false - otherwise
+ *         error - if value argument is not an object or prototype argument is not a function.
+ *
+ */
+jerry_value_t
+jerry_value_instanceof (const jerry_value_t value, /**< object value */
```
Comment (galpeter) [inline jerry-core/api/jerry.c]: I'm not sure if the `value` should be in the method name.

**Comment by LaszloLango** [comment]
Closing PR to prevent merging. Let's discuss and make decision on #2746 first.

**Pull request closed without merging**